### DISCUSSION TITLE
dind not working

### DISCUSSION THREAD

**Discussion by vquie** [renovatebot/renovate#19885] [Request Help] [answered]
Hi,

don't know if it's an issue or just misconfiguration. I am working with self-hosted renovate for a couple of days only. It was working flawlessly in the beginning. But since I switched to the slim image with dind I kind of never got any PR in my gitea. When i checked the logs it looked like I was rate limited at hub.docker.com. Since I sorted that out I kind of got the idea, that renovate should download images and run containers in the dind container, but it isn't. There are definitely updates pending in files it created PR before.

I am running renovate: 34.105.3 on k3s v1.26.0+k3s2.

My helm values:
```
---
cronjob:
  schedule: '15 * * * *'
image:
  repository: renovate/renovate
  tag: "34.105.3"
  pullPolicy: IfNotPresent
renovate:
  config: |
    {
      "platform": "gitea",
      "endpoint": "https://my-host/api/v1/",
      "token": "my-token",
      "autodiscover": true,
      "binarySource": "docker",
      "printConfig": true,
      "redis-url": "redis://my-redis:6379/0",
      "lockFileMaintenance":
        {
          "enabled": true
        },
      "hostRules": [
        {
          "hostType": "docker",
          "matchHost": "docker.io",
          "username": "my-username",
          "password": "process.env.DOCKER_HUB_PASSWORD"
        }
      ]
    }
dind:
  enabled: true
  slim:
    enabled: true
  image:
    repository: docker
    tag: 20.10.8-dind
serviceAccount:
  create: true
env:
  GITHUB_COM_TOKEN: "my-github-token"
  DOCKER_HUB_PASSWORD: "my-dockerhub-token"
  LOG_LEVEL: "info"
redis:
  enabled: true

```

**Comment by rarkins** [comment]
You probably need to run `docker login` so that `docker pull` isn't rate limited

**Comment by vquie** [comment]
As I said, I was rate limited but I am not limited anymore due to the login. There are no rate limit messages in the log. It logs things like `Failed to look up docker dependency` but without a reason.

**Comment by vquie** [comment]
I just tried with the big image and it does not work as well. So this is not just about dind.

I just downgraded the alpine/helm image in a drone ci pipeline and let renovate suggest the update. This is the debug output.

```
DEBUG: packageFiles with updates (repository=k3s/renovate, baseBranch=master)
       "config": {
         "droneci": [
           {
             "packageFile": ".drone.yml",
             "deps": [
               {
                 "depName": "alpine/helm",
                 "currentValue": "3.10.1",
                 "replaceString": "alpine/helm:3.10.1",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "docker",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "alpine/helm",
                     "message": "Failed to look up docker dependency alpine/helm"
                   }
                 ],
                 "versioning": "docker"
               }
             ]
           }
         ]
       }
```

**Reply by vquie** [reply]
All of a sudden renovate is working again. I created a new docker hub token (the previous one was new too) and updated the config. After a couple of failed runs I have seen that the new key has not been used at all. After trying different renovate image versions back and forth, halm uninstall and re-installs with the same config, it worked all of a sudden. Now I can also see that the key has been used on docker hub.

Is there a way to check if dind is working? The only PR I received where about helm values and docker compose files so far. Looking closely at the slim or dind image I can't see any further image pulled.

**Reply by rarkins** [reply]
To verify dind you would need to trigger an update which triggers a lock file update, e.g. npm, yarn, bundler, composer, poetry, etc.

**Reply by vquie** [reply]
And again it looks like it is not working. I just let the cron run two rounds (a bit more then an hour) without changing anything on renovate. I added drone ci files to other repos that were missing before though. I just received a PR as expected, but with the following message:

> ⚠ Dependency Lookup Warnings ⚠
> 
> Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.
> 
> Failed to look up docker dependency alpine/helm
> Files affected: .drone.yml

The corresponding part of the file looks like this.

```
steps:
  - name: helm update
    image: alpine/helm:3.10.2
    commands:
      - ...
```

Ratelimit-remaining is at `182;w=21600`. So I am not rate limited.

Something is not working as expected.

**Comment by vquie** [comment, marked as answer]
Since this is supposably not about dind I will create a new thread.
